api/edge_api/identities/views.py

```diff
@@ -245,3 +255,73 @@ def all(self, request, *args, **kwargs):
         )
 
         return Response(serializer.data)
+
+
+class EdgeIdentityWithIdentifierFeatureStateView(APIView):
+    permission_classes = [IsAuthenticated, EdgeIdentityWithIdentifierViewPermissions]
+    pagination_class = None
+
+    def initial(self, request, *args, **kwargs):
+        super().initial(request, *args, **kwargs)
+
+        serializer = EdgeIdentityIdentifierSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        identity_document = Identity.dynamo_wrapper.get_item(
+            f"{self.kwargs['environment_api_key']}_{serializer.data['identifier']}"
+        )
+
+        if not identity_document:
+            raise NotFound()
+        self.identity = build_identity_model(identity_document)
+
+    def _get_feature_state(
+        self, feature: typing.Union[str, int]
+    ) -> typing.Optional[EngineFeatureStateModel]:
+        def match_feature_state(fs):
+            if isinstance(feature, int):
+                return fs.feature.id == feature
+            return fs.feature.name == feature
+
+        feature_state = next(
+            filter(
+                match_feature_state,
+                self.identity.identity_features,
+            ),
+            None,
+        )
+
+        return feature_state
+
+    @swagger_auto_schema(
+        request_body=EdgeIdentityWithIdentifierFeatureStateRequestBody,
+        responses={200: EdgeIdentityFeatureStateSerializer()},
+    )
+    def post(self, request, *args, **kwargs):
+        feature = request.data.get("feature")
+        feature_state = self._get_feature_state(feature)
+        serializer = EdgeIdentityFeatureStateSerializer(
+            instance=feature_state, data=request.data, context={"view": self}
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response(serializer.data, status=200)
+
+    @swagger_auto_schema(
+        request_body=EdgeIdentityWithIdentifierFeatureStateRequestBody,
+        responses={200: EdgeIdentityFeatureStateSerializer()},
+    )
+    def put(self, request, *args, **kwargs):
```

Comment:
I wonder if we should _only_ allow `put`, right? By definition, PUT (I think) is create or update - POST is strictly create (although it's not something we have necessarily followed elsewhere in the application, admittedly).

Reply:
Yeah, I was thinking about that, but I went ahead with both because of 
> although it's not something we have necessarily followed elsewhere in the application, admittedly

Reply:
I'd ideally like to try and follow best practices on new endpoints though so can we do this?

Reply:
Yep, I removed the post